solr/core/src/java/org/apache/solr/core/backup/repository/BackupRepositoryUtil.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.core.backup.repository;
+
+import java.io.IOException;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.apache.lucene.util.IOUtils;
+import org.apache.solr.core.DirectoryFactory;
+
+/** Utility methods for {@link BackupRepository}. */
+public class BackupRepositoryUtil {
+
+  /**
+   * Copy a file from a source {@link Directory} to a destination {@link Directory} without
+   * verifying the checksum.
+   *
+   * @param sourceDir The source directory hosting the file to be copied.
+   * @param sourceFileName The name of the file to be copied.
+   * @param destDir The destination directory to copy the file to.
+   * @param destFileName The name of the copied file at destination.
+   */
+  public static void copyFileNoChecksum(
```

Comment:
[0] This doesn't seem too radically different from other methods that live on `BackupRepository` itself as "default" interface implementations or static methods...is there a particular reason you have this live in a separate 'util' class? (Maybe it's easier to test this way?)

I don't have an opinion on where this lives one way or another; I just want to make sure I understand how you've got things laid out...

Reply:
I thought this method was enough different from the others in the BR interface to not have its place there. And it is static, so I don't think it could be overridden. But now that you suggest, I'll move it inside BR.